### PR TITLE
FEATURE: Allow users to log in without access to the content module

### PR DESCRIPTION
**Pull request by daniellienert**
Before, it was not possible to log in to the Neos backend without having access to the content module, as the user was automatically redirected to the content module.

With this feature, an available module is determined when a user logs in and the user is redirected to that module.

# High level features

- Users no longer need to have access to the content module to be able to log in
- There is a setting `Neos.Neos.moduleConfiguration.preferredStartModules` available to define the preferred start modules. The first module of that list, which a user has access to is chosen for the start module. If none of the listed modules can be accessed, the first module of the list of accessible modules of a user is chosen:
- The backend modules and sub modules can now be sorted using the positional array sorter

```
Neos:
  Neos:
    moduleConfiguration:
      preferredStartModules: [ 'content', 'user/usersettings' ]
```

## Details

- A user who has no access to the content module is redirected to the first (preferred) accessible content module
- A user who has access to the content module is still redirected to the content module by default
- A user without access to the content module does not see the available sites menu
- When the `preferredStartModule` is not `content`, the content module is still selectable

# Update notes

- This removes `Neos.Neos.userInterface.routeAfterLogin`, it can be dropped from custom configuration

Resolves: #2351
Depends on: https://github.com/neos/neos-ui/pull/2793

### CODE REVIEW

File: Neos.Neos/Configuration/Settings.yaml [hunk 1]
```diff
@@ -66,15 +66,6 @@ Neos:
...
     userInterface:
-      # Sets the route after login. There will be one argument "node" set with the lastVisited or site node (and correct context)
-      # The below configuration sets this to the "old" UI route.
-      routeAfterLogin:
-        values:
-          '@package': 'Neos.Neos'
-          '@controller': 'Frontend\Node'
-          '@action': 'show'
-          '@format': 'html'
-
```
Comment (kdambekalns) [inline Neos.Neos/Configuration/Settings.yaml:77]: This should not be removed completely – the code still uses it (if set), so this must stay as the "definition" and documentation for the setting.

Comment (daniellienert) [reply Neos.Neos/Configuration/Settings.yaml:77]: Good catch @kdambekalns. The goal is still to get rid of it. Some more refactoring is needed though.

Comment (daniellienert) [reply Neos.Neos/Configuration/Settings.yaml:77]: Done. Haven't found any usages anymore.

**Review by markusguenther — APPROVED** [review]
Tried that locally with the neos-ui adjustment and it works nice.
Thanks for bringing that up. ❤️

**Comment by daniellienert** [comment]
@bwaidelich / @kdambekalns, I would love to hear you feedback on this.

**Review by kdambekalns — APPROVED** [review]
I like idea & concept. The code looks good to me (from reading), too.

**Pull request merged**